packages/core/src/vivliostyle/css-cascade.ts

```diff
@@ -2009,6 +2009,127 @@ export class ContentPropVisitor extends Css.FilterVisitor {
+    let leader: string = "";
+    if (values[0] instanceof Css.Ident) {
+      switch (values[0].stringValue()) {
+        case "dotted":
+          leader = ".";
+          break;
+        case "solid":
+          leader = "_";
+          break;
+        case "space":
+          leader = " ";
+          break;
+      }
+    } else if (values[0] instanceof Css.Str) {
+      leader = values[0].stringValue();
+    }
+    if (leader.length == 0) {
+      return new Css.Str("");
+    }
+    if (!this.postLayoutLeaderRegistered) {
+      Plugin.registerHook(
+        Plugin.HOOKS.POST_LAYOUT_BLOCK,
+        this.postLayoutLeader.bind(this),
+      );
+      this.postLayoutLeaderRegistered = true;
+    }
+    return new Css.Expr(
+      new Exprs.Native(
+        null,
+        (): Exprs.Result => {
+          return leader.replace(" ", "\u00A0"); // nbsp
+        },
+        "viv-leader",
+      ),
+    );
+  }
+  postLayoutLeaderRegistered: boolean = false;
+  postLayoutLeader: Plugin.PostLayoutBlockHook = (
+    nodeContext: Vtree.NodeContext,
+    checkPoints: Vtree.NodeContext[],
+    column: Layout.Column,
+  ) => {
+    // we want to access the bottom block element, which contains single leader().
+    if (nodeContext.viewNode.nodeType !== 1) {
+      return;
+    }
+    const container = nodeContext.viewNode as Element;
+    const leaders = container.querySelectorAll(`span>span[viv-leader]`);
+    if (leaders.length != 1) {
+      return;
+    }
+    const e = leaders[0];
+    const pseudoAfter = e.parentElement;
+    const leader = e.getAttribute("viv-leader-value");
+    const previous = e.textContent;
+
+    // reset the expanded leader
+    e.textContent = leader;
+    pseudoAfter.style.paddingTop = "0";
+    pseudoAfter.style.paddingLeft = "0";
+
+    const outer = column.clientLayout.getElementClientRect(container);
+    const innerInit = column.clientLayout.getElementClientRect(pseudoAfter);
+    // Some leader text ("_" e.g.) creates higher top than container.
+    const box = {
+      left: outer.left < innerInit.left ? outer.left : innerInit.left,
+      right: outer.right < innerInit.right ? innerInit.right : outer.right,
+      top: outer.top < innerInit.top ? outer.top : innerInit.top,
+      bottom: outer.bottom < innerInit.bottom ? innerInit.bottom : outer.bottom,
+    };
+    function overrun() {
+      const inner = column.clientLayout.getElementClientRect(pseudoAfter);
+      if (
+        box.left > inner.left ||
+        box.right < inner.right ||
+        box.top > inner.top ||
+        box.bottom < inner.bottom
+      ) {
+        return true;
+      }
+      return false;
+    }
+    function getPadding() {
+      const inner = column.clientLayout.getElementClientRect(pseudoAfter);
+      if (column.vertical) {
+        // Supporting only left-to-right, top-to-bottom here.
+        return outer.bottom - inner.bottom;
+      } else {
+        return outer.right - inner.right;
+      }
```

Comment:
縦書きのページの中で目次は横書きのレイアウトなどという場合を考慮すると、column.vertical ではなく、pseudoAfter要素に継承されている writing-mode プロパティで縦書きの判定をするのがよいと思います。また、rtlのサポートを追加したいです。次の修正でどうでしょう：

```suggestion
      const { writingMode, direction } =
        column.clientLayout.getElementComputedStyle(pseudoAfter);
      if (writingMode === "vertical-rl" || writingMode === "vertical-lr") {
        if (direction === "rtl") {
          return inner.top - outer.top;
        } else {
          return outer.bottom - inner.bottom;
        }
      } else {
        if (direction === "rtl") {
          return inner.left - outer.left;
        } else {
          return outer.right - inner.right;
        }
      }
```